Automattic/wp-calypso
# Packages: Migrate plans-grid to a new package, Plans Grid Next

## Proposed Changes

Here we are!

- we create a new package, `@automattic/plans-grid-next`, and migrate everything from  `plans-grid/**` there
- Storybook setup to add stories for all the exported components (obviously later)
- imports updated to be from this new package

Going forward, it's best if we do not import directly from the `/src` folder in this package, but rather export things from the main file `/src/index` and import directly from `@automattic/plans-grid-next`.

I chose to go with a new package (named `plans-grid-next`) instead of using the existing one (`@automattic/plans-grid`) as there are more imports from it in Calypso than I expected (there are more than what's exported from the main index). To keep things clean and have better control, we can deprecate that package gradually and eventually rename this one to be `plans-grid`.

## Testing Instructions



* Obviously everything `/plans/ [ site ]`, `/start/plan`, etc.
* Most everything should be caught by build and tests, and we've already done a directory cleanup in https://github.com/Automattic/wp-calypso/pull/86540. I think mostly on styles best to test to make sure there are no regressions.

## Pre-merge Checklist
